src/compiler/binder.ts

```diff
@@ -919,6 +934,16 @@ namespace ts {
                     checkStrictModeFunctionName(<FunctionExpression>node);
                     let bindingName = (<FunctionExpression>node).name ? (<FunctionExpression>node).name.text : "__function";
                     return bindAnonymousDeclaration(<FunctionExpression>node, SymbolFlags.Function, bindingName);
+
+                case SyntaxKind.CallExpression:
+                    // We're only inspecting call expressions to detect CommonJS modules, so we can skip
+                    // this check if we've already seen the module indicator
+                    if (isJavaScriptFile && !file.commonJsModuleIndicator) {
+                        bindCallExpression(<CallExpression>node);

```

Comment:
I would move the `!file.commonJsModuleIndicator` test into `bindCallExpression` so we have all the logic in one place.